regex-automata/src/nfa/thompson/builder.rs

```diff
@@ -91,10 +98,20 @@ enum State {
         /// The next state that this state should transition to.
         next: StateID,
     },
+    WriteLookaround {
+        lookaround_index: usize,
```

Comment:
We probably want to use `SmallIndex` here

Reply:
Yeah I thought about that as well